src/main/java/io/supertokens/passwordless/Passwordless.java

```diff
@@ -124,6 +129,121 @@ private static String generateUserInputCode() {
         return sb.toString();
     }
 
+    public static ConsumeCodeResponse consumeCode(Main main, String deviceId, String userInputCode, String linkCode)
+            throws RestartFlowException, ExpiredUserInputCodeException, IncorrectUserInputCodeException,
+            StorageTransactionLogicException, StorageQueryException, NoSuchAlgorithmException, InvalidKeyException {
+        PasswordlessSQLStorage passwordlessStorage = StorageLayer.getPasswordlessStorage(main);
+        long passwordlessCodeLifetime = Config.getConfig(main).getPasswordlessCodeLifetime();
+        int maxCodeInputAttempts = Config.getConfig(main).getPasswordlessMaxCodeInputAttempts();
+
+        String deviceIdHash;
+        String linkCodeHash;
+        if (linkCode != null) {
+            byte[] linkCodeBytes = Base64.decodeBase64URLSafe(linkCode);
+            byte[] linkCodeHashBytes = Utils.hashSHA256Bytes(linkCodeBytes);
+            linkCodeHash = Base64.encodeBase64String(linkCodeHashBytes);
```

Comment:
Can extract this into a function since it's being replicated in multiple places.